yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts

```diff
@@ -886,13 +886,21 @@ export class PXEOracleInterface implements ExecutionDataProvider {
     );
   }
 
-  public async removeNullifiedNotes(contractAddress: AztecAddress) {
+  /**
+   * Synchronizes notes associated with a given contract.
+   *
+   * Fetches notes from the NoteDataProvider and checks which nullifiers are present in the
+   * onchain nullifier Merkle tree -  up to the latest locally synced block. We use the
+   * locally synced block instead of querying the chain's 'latest' block to ensure correctness:
+   * notes are only marked nullified once their corresponding nullifier has been included in a
+   * block up to which the PXE has synced.
+   * This allows recent nullifications to be processed even if the node is not an archive node.
+   *
+   * @param contractAddress - The contract whose notes should be checked and nullified.
+   */
+  public async syncContractNotes(contractAddress: AztecAddress) {
```

Comment:
```suggestion
  public async syncActiveNoteNullifiers(contractAddress: AztecAddress) {
```

I don't like the original name because it would make me think that we are syncing new notes but we don't do that here - we load nullifiers for known active notes.

Not super sure my proposed name is good but I would say it might be because:
1. it having contract address on the input communicates that we are most likely doing that for a single contract,
2. it explicitly mentioning active notes communicates the function tries to load nullifiers only of known active notes.

But IDK

Reply:
Changed to syncNoteNullifiers